### ISSUE TITLE
[SQLsmith] ERROR:  cannot reference pseudo type pg_catalog.list

### ISSUE THREAD

**Issue by def-** [MaterializeInc/materialize#17870]
### What version of Materialize are you using?

v0.45.0-dev (f10e2c37f)

### How did you install Materialize?

Built from source

### What is the issue?

```
materialize=> SELECT CAST(NULL AS list) AS c5;
ERROR:  cannot reference pseudo type pg_catalog.list
```
Not sure if this is an expected limitation of how we currently implement lists. It's just a query error so I'll ignore it for now, but haven't seen this documented anywhere.

**Comment by def-** [comment]
Same for all other pseudo types like any, anyrange too.

**Comment by chaas** [comment]
I found this code
https://github.com/MaterializeInc/materialize/blob/a716d2b8b9a96b01e61859857822f4822e7d4baf/src/sql/src/plan/query.rs#L4960-L4965
in the function `scalar_type_from_catalog`, so I guess it's "intentional", but I'm not sure whether the plan was add this support later.

**Comment by benesch** [comment]
Yeah it's intentional for now, but PostgreSQL supports it so we should support it too eventually:

```
benesch=# select array[1]::anyarray;
 array 
-------
 {1}
(1 row)
```
